src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
+
+    static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= LOAD_FACTOR * table.length) {
+            resize();
+        }
+        int hash = hash(key);
+        Node<K, V> newNode = new Node(hash, key, value, null);
+        Node<K, V> node = table[hash];
+        if (node == null) {
+            table[hash] = newNode;
+        } else {
+            while (true) {
```

Comment:
`while (true)` smells bad, refactor it. u should do it while next is available, once it not found, u finish the loop and assign new node there.

Reply:
i had try to do with optional, but haven`t much time.